src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataResource.java

```diff
@@ -185,4 +185,12 @@ ResponseEntity updateCaseDataValue(@PathVariable UUID caseUUID, @PathVariable St
         CaseData caseData = caseDataService.getCaseDataByReference(decodedRef);
         return ResponseEntity.ok(GetCaseResponse.from(caseData, true));
     }
+
+    @Authorised(accessLevel = AccessLevel.READ)
+    @GetMapping(value = "/case/reference/{caseUUID}")
+    ResponseEntity<GetCaseReferenceResponse> getCaseReference(@PathVariable UUID caseUUID) {
```

Comment:
We should cache this method using spring cache. Also we should cache null responses (most caching annotations we have do exclude null result caching). This will help to minimise the non case uuid requests.

Reply:
Agreed